commands/perks.js

```diff
@@ -0,0 +1,114 @@
+/**
+ * Display EQ and EQ2 Perks
+ *
+ * @example   !perks
+ */
+exports.run = async (client, message, args, level) => { // eslint-disable-line no-unused-vars
+
+    // category mappings, to add more copy and paste the text from the faction page
+    // do not use the wording on the torn wiki page as some entries are worded differently
+    // this is not a complete list and will need to be adjusted moving forward as eq or
+    // eq2 unlock/leverage other branches.
+    var categories = {
+        'Toleration' : [
+            'Reduces drug addiction gain by',
+            'Decreases risk of overdose by',
+            'Reduces passive negatives from drugs and addiction by',
+        ],
+
+        'Voracity' : [
+            'Increases nerve gain from alcohol by',
+            'hours of maximum booster cooldown',
+            'Increases energy gain from energy drinks by',
+            'Increases happy gain from candy by',
+        ], 
+
+        'Suppression' : [
+            'upon escape attempt',
+            'Increases dexterity passively by',
+            'Increases defense passively by',
+            'Increases maximum life by',
+        ],
+
+        'Steadfast' : [
+            'Increases strength gym gains by',
+            'Increases speed gym gains by',
+            'Increases defense gym gains by',
+            'Increases dexterity gym gains by',
+        ],
+
+        'Excursion' : [
+            'Increases maximum traveling capacity by',
+            'Reduces travelling fees by',
+            'Increases hunting income by',
+            'Reduces rehabilitation costs by',
+            'ncreases cayman bank interest by',
+        ]
+    };
+
+    var urls = []
+
+    // generate a list of urls for faction name and faction perk api calls
+    Object.keys(client.auth.factionApiKeys).forEach(faction_key => {
+        var api_key = client.auth.factionApiKeys[faction_key];
+        urls.push(`https://api.torn.com/user/?selections=perks,profile&key=${api_key}`);
+    });
+
+    // list of results for all factions defined in client.auth.factionApiKeys
+    const results = await Promise.all(urls.map((url) => fetch(url).then((r) => r.json()))).catch((error) => {});
+    results.forEach(result => {
+        perk_map = {}
+
+        // for each perk, check to see if a substring from any of the categories
+        // match, if they do add it to perk_map in the structure of: k = category, 
+        // v = perk
+        result['faction_perks'].forEach(perk => {
+            Object.keys(categories).forEach(category_name => {
+
+                // if the perk_map mapping does not exist, create it
+                if(!perk_map.hasOwnProperty(category_name)) {
+                    perk_map[category_name] = [];
+                }
```

Comment:
I would move lines 68-71  and insert at line 76 inside the actual category mapping below rather than building all the categories. Then you only stub out a category if it's needed.

But honestly doesn't really matter. This isn't a consumer app where we are trying to optimize every operation.